tests/conftest.py

```diff
@@ -396,3 +401,111 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="session")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg') and config.RUN['google_api_secret']:
+        log.info(f"Tests will be running while IO is in the background")
+
+        request.results = list()
+        request.temp_file = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='test_status', delete=False
+        )
+
+        def get_test_status():
+            with open(request.temp_file.name, 'r') as t_file:
+                return t_file.readline()
+
+        def set_test_status(status):
+            with open(request.temp_file.name, 'w') as t_file:
+                t_file.writelines(status)
+
+        set_test_status('running')
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            set_test_status('finished')
+            try:
+                for status in TimeoutSampler(90, 3, get_test_status):
+                    if status == 'terminated':
+                        break
+            except TimeoutExpiredError:
+                log.warning(
+                    "Background IO was still in progress before IO "
+                    "thread termination"
+                )
+
+            request.thread.join(timeout=30)
+            log.info(f"Background IO has stopped")
```

Comment:
This is not true. The timeout in `join()` call just makes it wait given amount of time for the thread to finish (to prevent blocking the code if the thread won't stop itself). You don't know if the thread is still running or not after such call, see:

> When the timeout argument is present and not None, it should be a floating point number specifying a timeout for the operation in seconds (or fractions thereof). As join() always returns None, you must call is_alive() after join() to decide whether a timeout happened – if the thread is still alive, the join() call timed out.

https://docs.python.org/3.6/library/threading.html#threading.Thread.join

Reply:
removed the TO

Reply:
ok, either this or using `is_alive()` to tweak the log messages accordingly will work